packages/langium/src/generator/generator-node.ts

```diff
@@ -4,31 +4,75 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
-const EOL = (typeof process === 'undefined') ? '\n' : (process.platform === 'win32') ? '\r\n' : '\n';
+import { processGeneratorNode } from './node-processor';
 
-export type GeneratorNode = CompositeGeneratorNode | IndentNode | NewLineNode | string;
+export const EOL = (typeof process === 'undefined') ? '\n' : (process.platform === 'win32') ? '\r\n' : '\n';
+
+export type Generated = GeneratorNode | string | undefined;
+export type GeneratorNode = CompositeGeneratorNode | IndentNode | NewLineNode;
+
+export function isGeneratorNode(node: unknown): node is CompositeGeneratorNode | IndentNode | NewLineNode {
+    return node instanceof CompositeGeneratorNode
+        || node instanceof IndentNode
+        || node instanceof NewLineNode;
+}
+
+export function isNewLineNode(node: unknown): node is NewLineNode {
+    return node instanceof NewLineNode;
+}
+
+export function toString(object: unknown): string {
+    if (isGeneratorNode(object))
+        return processGeneratorNode(object);
+    else
+        return String(object);
+}
 
 export class CompositeGeneratorNode {
```

Comment:
Please add comments to the methods here to explain their use cases.

Reply:
Done!